### PR TITLE
django-nextjs package integration 

### PR DESCRIPTION
**Pull request by LaurenceZanotti**
Continuation of PR #22 in a separate branch for testing reasons. If the `django-nextjs` is successfully added and attends the project's needs, it will be merged to the #22 branch and then continue the feature implementation from there.

This approach should fix session persistance, CORS and CSRF errors.

### CODE REVIEW

**Comment by LaurenceZanotti** [comment]
Must create tests for the whole authentication process, both server and client side.

**Comment by LaurenceZanotti** [comment]
Wrote tests for register action but it worked only once because the test database is keeping it's records. Thats because `docker-compose.test.yml` tests run command has `--keep-db` flag, which is a workaround for errors mentioned within the issue #25. 

Need to find a way to create and destroy databases (a possible fix was mentioned at #25), as the test assumes the database will be destroyed. Or the newly created user should be manually deleted from the database after the end of the test.

Solving issue #25 might solve issue #6 regarding CI/CD errors too.

Committing register test mentioned after this comment.

**Comment by LaurenceZanotti** [comment]
After investigating the test commited above at [1e128c1](https://github.com/LaurenceZanotti/cs50web-capstone/pull/23/commits/1e128c10566dcc84378fa2017539f77b3ca466bb), found the following symptons (most likely related to #25 and #6):

Symptoms: `test_register_jobseeker` ran once successfully, then failed with "User already exists"

Possible cause: For some reason the user is not being deleted

Investigation of user not being deleted:

- Database is not being destroyed because of `--keepdb`
- Selenium Client (or more precisely, the api django app container) was creating a user at the default database

Bug: Django Client was using the test database, while the Selenium was using the default database, confirmed after visiting the `/admin` area.

Cause: Django Client uses the api routes directly, but Selenium Client interacts with NextJS generated HTML, sending requests to the api container, which is (accidentaly) set to use the default database.

**Comment by LaurenceZanotti** [comment]
Breaking changes to test enviroment on `docker-compose.test.yml` and environment vars must be reviewed.

**Comment by LaurenceZanotti** [comment]
What fixed the bug mentioned:

- Created a copy of `initial_data.json` fixture and changed users's usernames. (See `test_initial_data.json`)
- Changed `api` container running command to `python manage.py testserver --addrport 0.0.0.0:5000 test_initial_data.json --noinput`
- Changed `api` database user settings to `DB_TEST_USER` by enabling `IS_CICD_TESTING` env var (see database `settings.py`)
- Added `--noinput` flag to both `api` and `test` Django manage.py commands to enable database destruction and recreation after each tests, resetting data
- Removed `--keepdb` flag from both `api` and `test` Django manage.py commands

In other words:

Due to `--noinput` flag in both api and test containers, it avoids user input prompts automatically saying yes to proceed, then:

- The api destroys and creates any test database it finds, **then starts listening to requests**
- The test container destroys and creates any test database it finds, **then starts testing**

Test data **is not mixed** with default data because of `test_initial_data.json` fixture.

`manage.py testserver` command makes sure that api is using the test database.

Those changes fixed the **api container not using the test database during Selenium client testing**

**Comment by LaurenceZanotti** [comment]
On GH Actions `api` container ends with Connection refused using host "mariadatabase" (Docker internal DNS name).
